### DISCUSSION TITLE
Wayfire 0.9.0 - question about packages dependencies

### DISCUSSION THREAD

**Discussion by pondichys** [void-linux/void-packages#52487] [Q&A] [answered]
Greetings,

I want to update **Wayfire** to version **0.9.0** and I have some question about how the current package and a specific dependency are currently configured.

The **Wayfire** project depends on 4 external components
- wlroots
- wf-utils
- wf-touch
- wf-config

Those components are defined in the `subprojects` directory as git modules.
The current **Wayfire** template do pull `wf-utils` and `wf-touch` but not `wlroots` neither `wf-config` as they are available in their own separated Void package.

The issue I encounter is that to be able to compile **Wayfire 0.9.0**, it requires the version **0.9.0** of the `wf-config` component. It seems that `wf-config`, as both `wf-tools` and `wf-touch`, is a required dependency of **Wayfire** As such, I find it a bit odd that it is defined in its own package.

My thoughts are:

- we can respect the original author's view and keep `wf-config` as a separated package. Is there a way to put the update of `wf-config` and of **Wayfire** in the same PR and be sure the build will happen in the correct sequence ? If not, then I can create 2 PRs that will require a specific order of merging but I ifnd it tricky.
- we can modify the **Wayfire** package to include `wf-config` the same way `wf-touch` and `wf-utils` are handled and delete the separate `wf-config` package. My reasoning is that I don't see a good reason to install wf-config with another wayland compositor than **Wayfire**. I would also need some guidance to do so.

What do you think ? Do you see another way to handle that ?

Thanks a lot for your valuable input.

Regards,
Seb

**Comment by classabbyamp** [comment, marked as answer]
wf-utils and wf-touch do not have releases of their own, and are used only by wayfire itself, so they must be in the wayfire template. wf-config has its own releases so it goes in its own template. just update them at the same time as 2 commits in one pr
